crates/fluvio-version-manager/src/common/settings.rs

```diff
@@ -0,0 +1,142 @@
+use std::fs::write;
+use std::path::PathBuf;
+
+use anyhow::{Error, Result};
+use serde::{Deserialize, Serialize};
+use semver::Version;
+
+use fluvio_hub_util::fvm::Channel;
+
+use super::workdir::fvm_workdir_path;
+
+pub const SETTINGS_TOML_FILENAME: &str = "settings.toml";
+
+/// The `settings.toml` is in charge of keeping track of the active version
+/// through the default key, which holds the name of the directory under
+/// `~/.fvm/pkgset/default/versions` for the desired default version.
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Settings {
+    /// The active `channel` for the Fluvio Installation
+    pub channel: Option<Channel>,
+    /// The specific version in use
+    pub version: Option<Version>,
+}
+
+impl Settings {
+    /// Used to create an empty `settings.toml` file. This is used when the user
+    /// installs FVM but no version is set yet.
+    pub fn init() -> Result<Self> {
+        let settings_path = Self::settings_file_path()?;
+
+        if settings_path.exists() {
+            return Err(Error::msg(format!(
+                "Settings already exists on {}",
+                settings_path.display()
+            )));
+        }
+
+        let initial = Self {
+            channel: None,
+            version: None,
+        };
+
+        initial.save()?;
```

Comment:
Is there a reason this needs to be set to empty? why not save when channel is set

Reply:
This also could be lazily loaded like fluvio config

Reply:
Mmm yeah is part of setup files, so `fvm install` is not responsible of checking the existence of `settings.toml`. The other approach that comes to my mind would involve checking if the `settings.toml` exists and then write otherwise create.